NotifyTests/IntegrationTests/NotificationIntegrationClientTests.cs

```diff
@@ -170,12 +173,29 @@ public void GetAllNotifications()
 
 			foreach (Notification notification in notifications)
 			{
-				AssertNotification(notification);
+				NotifyAssertions.AssertNotification(notification);
 			}
 
 		}
 
 		[Test, Category("Integration")]
+		public void GetReceivedTexts()
+		{
+			NotificationClient client_inbound = new NotificationClient(NOTIFY_API_URL, INBOUND_SMS_QUERY_KEY);
+			ReceivedTextListResponse receivedTextListResponse = client_inbound.GetReceivedTexts();
+			Assert.IsNotNull(receivedTextListResponse);
+			Assert.IsNotNull(receivedTextListResponse.receivedTexts);
+			Assert.IsTrue(receivedTextListResponse.receivedTexts.Count > 0);
+
+			List<ReceivedTextResponse> receivedTexts = receivedTextListResponse.receivedTexts;
+
+			foreach (ReceivedTextResponse receivedText in receivedTexts)
```

Comment:
I'm not a fan of Logic in tests.

receivedTexts = receivedTexts.OrderBy(o=>o.<the thing to order by>).ToList();

then NotifyAssertions.AssertReceivedTextResponse(receivedTexts.getIndex(0))

Reply:
Not sure if it requires ordering as I'm not checking the content of each one response, I'm just checking that each response is a valid `ReceivedText` object. Could just check that the first one is valid

Reply:
Its just difficult to debug if one fails you are not sure if its message 1 or message 5 etc where as splitting it out you know.

Reply:
Sure, can put this as tech debt, I'm just using the same pattern for testing consistency but perhaps should compare the tests across all the clients at some stage to ensure they are covering the same ground?